core/server/worker/src/main/java/alluxio/worker/grpc/AbstractReadHandler.java

```diff
@@ -484,7 +484,6 @@ private void replyError(Error error) {
     private void replyEof() {
       mSerializingExecutor.execute(() -> {
         try {
-          Preconditions.checkState(!mContext.isDoneUnsafe());
```

Comment:
Why is this the case now?

Reply:
we shouldn't need to throw an error if the context is already done (sent a stream-ending response back to the client). Rather than throw the exception, we just log it now in case that we do hit this and want to debug further. It is uncommon, but does happen from time to time.